### PR TITLE
84-homework-student-feature-use-the-saved-completions-time-to-generate-stats-and-averages

### PR DESCRIPTION
**Pull request by AllanVester**
## Description

A new button has been added to the dashboard. This button, when clicked, shows the user a series of stats. These are calculated using the database whenever block_homework is loaded. It also calculates the user's reading speed, in minutes per page, and then it uses this to estimate the time needed for anything with a page count. Video times are also used for estimates. The reading speed is based on a default of 2 minutes per page, with the user's statistics gradually influencing it, weighted such that after 3 hours, the user's speed becomes more important than the default.

The estimates are shown for each material when clicking "time", but they are also shown for the entire homework instance.

## Walkthrough

As long as you have any homework available, you can see the estimate on the dashboard and by clicking time.

Dashboard:
image

Time:
image

The stats page will have every stat at 0 unless you have at least one completion.
image

### CODE REVIEW

File: server/moodle/blocks/homework/version.php
Comment (Verdoner) [inline server/moodle/blocks/homework/version.php:1]: Kinda silly to not update the version to the date of when changes were implemented

Comment (sinagaming69) [reply server/moodle/blocks/homework/version.php:1]: ups

Comment (AllanVester) [reply server/moodle/blocks/homework/version.php:1]: Done

Comment (sinagaming69) [reply server/moodle/blocks/homework/version.php:1]: ændret

**Comment by sinagaming69** [comment]
> It seems as if I can not have a 100% completion even though I have no more unfinished homework.
> 
> image
> 
> If this is an issue on my end then nevermind. The rest looks good.

Vil du dele præcis hvilke homework materials og completions der var, for at den fejl dukker op? Vi kan ikke genskabe den

**Comment by Verdoner** [comment]
Here you can see the homeworks I have:
image

The material associated with that homework:
image

Which materials are marked as completed:
image

From this information it would seem to me that the file associated to the homework with id = 1 is not completed, yet I have no option of actually completing it as seem here: 

image

**Comment by sinagaming69** [comment]
Vi kan se fejlen. Det sker i get_infohomework_modal. Det er fordi den kode, der deler lektier op i literature, link eller video, for at tildele dem ikoner, jo så ender med at skippe de lektier, der ikke passer ind i nogen af de tre kategorier. Ved ikke om det så skal laves som en bugfix, da der skulle laves en hel tredje kategori i forhold til ikoner osv.

**Comment by Verdoner** [comment]
Hvis det ikke har noget med jeres opgave at gøre, så tænker jeg ikke umiddelbart at det er noget i skal fikse nu (medmindre i har lyst). Jeg ville nok oprette et nyt issue med det i stedet.

**Review by Verdoner — APPROVED** [review]

**Pull request merged**